### PR TITLE
Add endorsements

### LINKED ISSUE #103: Add endorsements
e.g.

https://x.com/schickling/status/1832361798130954335
https://x.com/raymmar_/status/1832464000996548806
https://x.com/xdamman/status/1837021472499343492

Others?

Let's also include ecosystem partner endorsements, e.g., OSI, OC, Ruby Central, etc.

Should rotate one on the homepage and collect multiple on an /endorsements page.

### PR DESCRIPTION
**Pull request by vladh**
Closes #103.

Currently, 3 out of 4 endorsements are shown randomly.

I've followed [your suggestion to shorten the quotes](https://github.com/opensourcepledge/opensourcepledge.com/issues/103#issuecomment-2346299172), and made minor punctuation edits and slightly shortened Raymmar's quote so that everything fits our container width nicely.

### CODE REVIEW

File: src/components/Endorsements.astro [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    sourceUrl="https://x.com/raymmar_/status/1832464000996548806"
+  >
+    <p>Great initiative. [Vital] for the next step of internet growth. 💪</p>
+  </Endorsement>
+  <Endorsement
+    name="Andreas Klinger"
+    handle="@andreasklinger"
+    imageUrl="/images/endorsers/andreasklinger.webp"
+    sourceUrl="https://x.com/andreasklinger/status/1832365299686916531"
+  >
```
[the remark this answers is not in the document]
Comment (vladh) [reply src/components/Endorsements.astro]: OK!

**Comment by vladh** [comment]
Rebased to squash the Andreas endorsement + image out.

File: src/components/Endorsements.astro [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    <p>I can't overstate how excited I am about this. ❤️</p>
+  </Endorsement>
+  <Endorsement
+    name="Raymmar"
+    handle="@raymmar_"
+    imageUrl="/images/endorsers/raymmar.webp"
+    sourceUrl="https://x.com/raymmar_/status/1832464000996548806"
+  >
+    <p>Great initiative. [Vital] for the next step of internet growth. 💪</p>
```
Comment (chadwhitacre) [inline src/components/Endorsements.astro:23]: "Much needed and vital" didn't work because of word-wrapping, or ... ?

Comment (vladh) [reply src/components/Endorsements.astro:23]: Yeah the 💪🏻 emoji ended up being the only thing on the second line. I can try something else if you think the full sentence would be better.

Comment (vladh) [reply src/components/Endorsements.astro:23]: (Could just remove the emoji)

File: src/components/Endorsements.astro [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  </Endorsement>
+</div>
+
+<style>
+</style>
+
+<script>
+  import { shuffle, range } from "../util.ts";
+
+  const N_TO_CHOOSE = 3;
```
Comment (chadwhitacre) [inline src/components/Endorsements.astro:44]: Can we drop to one on homepage and make a page at `/endorsements` with all posts? "EXPLORE ALL ENDORSEMENTS" link similar to "EXPLORE ALL COMPANIES"

File: src/components/Endorsements.astro (hunk 2 above)
Comment (chadwhitacre) [reply src/components/Endorsements.astro:23]: Yeah let's do that

**Comment by vladh** [comment]
Screen Shot 2024-10-05 at 00 45 15

File: src/pages/endorsements.astro [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    <Blob kind="grad-dots-06" top="37%" right="-5rem"></Blob>
+  </div>
+  <div class="container">
+    <Blob kind="solid-fill-05" top="20%" right="-16rem"></Blob>
+    <Blob kind="solid-fill-04" top="90%" left="-20rem"></Blob>
+    <Blob kind="grad-dots-03" top="10%" left="-15rem"></Blob>
+
+    <main id="main-content">
+      <h1>Endorsements</h1>
+      <p>See what lovely folks from across the Open Source ecosystem had to say about the Open Source Pledge.</p>
```
Comment (chadwhitacre) [inline src/pages/endorsements.astro:23]: Tighten up this way? We have @ThePledge on X, are we ready to start referring to ourselves this way sometimes on the website?

```suggestion
      <p>See what lovely folks across the Open Source ecosystem have to say about the Pledge.</p>
```

File: src/pages/endorsements.astro [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+      <section>
+        <EndorsementList full={true}></EndorsementList>
+      </section>
+
+      <section class="flex-center bottom-cta">
+        <div>
+          <div class="text-center">
+            <h2>Have they convinced you?</h2>
```
Comment (chadwhitacre) [inline src/pages/endorsements.astro:32]:
```suggestion
            <h2>Convinced?</h2>
```

File: src/components/EndorsementList.astro [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+import Endorsement from "./Endorsement.astro";
+
+interface Props {
+  full?: boolean;
+}
+
+const { full } = Astro.props;
+---
+
+<div class="endorsement-list" data-full={full}>
```
Comment (chadwhitacre) [inline src/components/EndorsementList.astro:14]: I guess we should sort by impact, eh? Or maybe next week we go with sections?

1. OSI at the top
2. Other foundations/institutions, sorted by date (desc?)
3. Individuals, sorted by date (desc?)

Comment (chadwhitacre) [reply src/components/EndorsementList.astro:14]: This is fine for today. 👍

**Comment by vladh** [comment]
@chadwhitacre Thanks for doing these fixes! Sure thing, we can revisit ordering etc once we have more endorsements.

**Comment by chadwhitacre** [comment]
The site is looking better and better! Awesome work! :-)

**Pull request merged**